rosetta-node/docker-compose.yaml

```diff
@@ -0,0 +1,324 @@
+            level: warn
+          - value: Checkpointer
+            key: logger
+            level: warn
+          default: info
+      EOF
+
+  # Shut down chainweb-node before running this.
+  chainweb-validate-db:
+    container_name: chainweb-validate-db
+    restart: "no"
+    image: ghcr.io/kadena-io/chainweb-node/ubuntu:latest
+    environment:
+      KADENA_NETWORK: ${KADENA_NETWORK:-mainnet01}
+    stop_signal: SIGINT
+    profiles:
+    - validate-db
+    ulimits:
+      nofile:
+        soft: 65535
+        hard: 65535
+    depends_on:
+      chainweb-validate-db-config:
+        condition: "service_completed_successfully"
+    volumes:
+    - chainweb-db:/chainweb/db:rw
+    - chainweb-config:/config:ro
+    entrypoint: /bin/bash
+    command:
+    - "-c"
+    - |
+      [ -d "/chainweb/db/0/rocksDb" ] || { echo "missing database. Try running chainweb-validate-db-sync first" 1>&2; exit 1; }
+      rm -rf "/db/0/sqlite/*"
+      /chainweb/chainweb-node \
+        +RTS \
+        -T \
+        -H1G \
+        -A64M \
+        -RTS \
+        --config-file=/config/chainweb-validate-db-config.yaml \
+        --log-handle=stderr
+
+  # ########################################################################## #
+  # Chainweb Node
+
+  chainweb-node:
+    container_name: chainweb-node
+    image: ghcr.io/kadena-io/chainweb-node/ubuntu:latest
+    environment:
+      KADENA_NETWORK: ${KADENA_NETWORK:-mainnet01}
+    restart: unless-stopped
+    stop_grace_period: "20s"
+    stop_signal: SIGINT
+    ulimits:
+      nofile:
+        hard: 65535
+        soft: 65535
+    ports:
+    - mode: host
+      protocol: tcp
+      published: 1789
+      target: 1789
+    - mode: host
+      protocol: tcp
+      published: 1848
+      target: 1848
+    volumes:
+    - chainweb-db:/chainweb/db:rw
+    deploy:
+      restart_policy:
+        condition: on-failure
+        delay: "5s"
+        max_attempts: 3
+        window: "120s"
+      update_config:
+        delay: "60s"
+        order: stop-first
+    entrypoint:
+    - /chainweb/chainweb-node
+    command:
+    - +RTS
+    - -T
+    - -H1G
+    - -A64M
+    - -RTS
+    - --database-directory=/chainweb/db
+    - --chainweb-version=${KADENA_NETWORK:-mainnet01}
+    - --log-level=warn
+    - --log-handle=stderr
+    - --telemetry-log-handle=stderr
+    - --rosetta
+    - --p2p-port=1789
+    healthcheck:
+      interval: "1m"
+      retries: 3
+      start_period: "30s"
+      test:
+      - CMD
+      - /bin/bash
+      - "-c"
+      - "set -e; while : ; do /bin/bash -c \"$$1\" ; shift || break ; done"
+      - '--'
+      - |
+        # check health endpoint
+        exec 3<>/dev/tcp/localhost/1848
+        printf "GET /health-check HTTP/1.1\r\nhost: http://localhost:1848\r\nConnection: close\r\n\r\n" >&3
+        grep -q "200 OK" <&3 || exit 1
+      timeout: "10s"
+
+  # ########################################################################## #
+  # Container Monitor
+
+  container-monitor:
```

Comment:
If a container is stopped, is the monitor container supposed to be stopped as well?

Reply:
That should not be necessary. It only restarts containers that are not healthy.